src/frontend/src/lib/utils/credentials.utils.ts

```diff
@@ -0,0 +1,19 @@
+import type { UserProfile } from '$declarations/backend/backend.did';
+import { isNullish } from '@dfinity/utils';
+
+/**
+ * Returns true if the user has a verified proof of uniqueness credential
+ * Returns false if the user has no verified proof of uniqueness credential
+ * Returns undefined if the user has no profile yet. It means it's loading.
+ *
+ * @param profile {UserProfile | null | undefined} The user profile.
+ * @returns {boolean | undefined}
+ */
+export const hasPouhCredential = (profile: UserProfile | null | undefined): boolean | undefined => {
+	if (isNullish(profile)) {
+		return undefined;
+	}
+	return profile.credentials.some(
+		(cred) => 'ProofOfUniqueness' in cred.credential_type && cred.verified_date_timestamp.length > 0
```

Comment:
should the partial edge case be handled? i.e. a user has `'ProofOfUniqueness'` but no `verified_date_timestamp`?

Reply:
It's handled this case, isn't it?

The value of `verified_date_timestamp` is an optional timestamp.

If `verified_date_timestamp` is not present, it's an empty array. Therefore, cred.verified_date_timestamp.length > 0 will be false  and `hasPouhCredential` will be false.

There is a test with this scenario.

Reply:
I see, thanks for the details. It might improve readability to also have an `isValidCredentials` (or something) method. I might just be me of course but, I didn't understand that checking for the presence of `verified_date_timestamp` was the authoritative parameter to determine if a credential is valid or not.

Reply:
Good point, it will help readability. Done